tests/integration/test_upgrade.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import logging
+
+import pytest
+import requests
+from pytest_operator.plugin import OpsTest
+
+from conftest import fetch_charm, get_charm_resources
+from helpers import ACTIVE_STATUS, APP_NAME
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.abort_on_fail
+@pytest.mark.usefixtures("deploy_current_stable")
+class TestUpgrade:
```

Comment:
Also, why are we wrapping this in a test class?

Reply:
I tried to keep it as is was before (for the normal deploy test structure). This way we can add more tests function to the class, and I suppose, use the same fixture.

Reply:
I don't think the test class gives us much. Also it's interesting to note how the pytest-operator affects tests - https://github.com/charmed-kubernetes/pytest-operator/blob/main/docs/reference.md#fixtures
With a module level fixture all the tests in the same file already use the same fixture. One thing I haven't figured out is why tests seem to run sequentially, for example the Grafana-k8s operator tests look like this https://github.com/canonical/grafana-k8s-operator/blob/main/tests/integration/test_external_url.py

Reply:
Fair enough. I didn't notice the module-scope behavior. I'll remove the classes.